src/provider/provider_tracking.c

```diff
@@ -382,6 +382,12 @@ static umf_result_t trackingFree(void *hProvider, void *ptr, size_t size) {
     ret = umfMemoryProviderFree(p->hUpstream, ptr, size);
     if (ret != UMF_RESULT_SUCCESS) {
         LOG_ERR("upstream provider failed to free the memory");
```

Comment:
in case the free() is not supported we should call LOG_ERR. Please do not nest ifs here and use switch or if/else if

Reply:
Removed (code change). It will be fixed in another way.
Done.

Reply:
Code changed.
Done.